l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -125,9 +123,10 @@ contract ExecutorFacet is Base, IExecutor {
         // See SystemLogKey enum in Constants.sol for ordering.
         uint256 processedLogs;
 
-        // #if VALIDIUM_MODE == false
-        bytes32 providedL2ToL1PubdataHash = keccak256(_newBatch.totalL2ToL1Pubdata);
-        // #endif
+        bytes32 providedL2ToL1PubdataHash;
+        if (s.feeParams.pubdataPricingMode == PubdataPricingMode.Rollup) {
+            providedL2ToL1PubdataHash = keccak256(_newBatch.totalL2ToL1Pubdata);
+        }
```

Comment:
Let's make it foolproof, 
```suggestion
        bytes32 providedL2ToL1PubdataHash;
        if (s.feeParams.pubdataPricingMode == PubdataPricingMode.Rollup) {
            providedL2ToL1PubdataHash = keccak256(_newBatch.totalL2ToL1Pubdata);
        } else if (s.feeParams.pubdataPricingMode == PubdataPricingMode.Validium) {
            require(_newBatch.totalL2ToL1Pubdata.length == 0, "Mts");
        } else {
            revert("sop");
        }
```

Reply:
If I'm not mistaken, because of how pubdata is being encoded today, the expected length of `            _newBatch.totalL2ToL1Pubdata.length` should be 1 (the lengths of every field concatenated to the pubdata is part of `totalL2ToL1Pubdata`, and if you send no pudbata you get `Bytes([0])`).

Reply:
Addressed in [abb786d](https://github.com/matter-labs/era-contracts/pull/225/commits/abb786d50645b74150d85bb80e97d168b686d05d)